lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -445,7 +471,14 @@ def get_students_opened_subsection(request, csv=False):
     students = models.StudentModule.objects.select_related('student').filter(
         module_state_key__exact=module_state_key,
         module_type__exact='sequential',
-    ).values('student__username', 'student__profile__name').order_by('student__profile__name')
+    ).values('course_id','student__id','student__username', 'student__profile__name').order_by('student__profile__name')
+
+    # is there a way to join student_courseaccessrole and courseware_studentmodule
+
+    # here we are using the first record to get course_id.
+    course_key = locator.CourseLocator.from_string(students[0]['course_id'])
```

Comment:
I think the has a couple of issues.  First, what would happen if `student` were `[]`?  You'd have an uncaught exception, so you'd minimally need a `try...except` block to handle that.

But I actually think fetching the `course_id` from the database is probably the wrong approach.  The client-side page that issues this request must know which course it's on (it's at least in the URL), so let's just have it send course_id as a parameter in the request.  (If the client-side page doesn't have course_id/course_key readily available, you can have it be added to the template variable it the rendering view).  I think @dcadams is familiar with this part of the code and can help.